### PR TITLE
APIS-4724 Subscription Fields domain refactoring

### PR DESCRIPTION
**Pull request by adampridmore**
Change the service layer domain to match the improvements made in gatekeeper recently.

### CODE REVIEW

File: app/connectors/ApiSubscriptionFieldsConnector.scala [hunk 1]
```diff
@@ -17,125 +17,197 @@
...
-  def fetchFieldValues(clientId: String, apiContext: String, apiVersion: String)(implicit hc: HeaderCarrier): Future[Option[SubscriptionFields]] = {
-    val url = urlSubscriptionFieldValues(clientId, apiContext, apiVersion)
-    Logger.debug(s"fetchFieldValues() - About to call $url in ${environment.toString}")
-    retry {
-      http.GET[SubscriptionFields](url).map(Some(_))
-    } recover recovery(None)
+  def fetchFieldValues(clientId: String, context: String, version: String)(implicit hc: HeaderCarrier): Future[Seq[SubscriptionFieldValue]] = {
+
+    def getDefinitions() =
+      fetchFieldDefinitions(context, version)
```
Comment (peteslater-ee) [inline app/connectors/ApiSubscriptionFieldsConnector.scala]: Pls make it one line

File: project/metals.sbt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// DO NOT EDIT! This file is auto-generated.
+// This file enables sbt-bloop to create bloop config files.
+
+addSbtPlugin("ch.epfl.scala" % "sbt-bloop" % "1.4.0-RC1-105-118a551b")
```
Comment (peteslater-ee) [inline project/metals.sbt]: Probably don't need that in here.

File: app/connectors/ApiSubscriptionFieldsConnector.scala (hunk 1 above)
Comment (AndySpaven) [reply app/connectors/ApiSubscriptionFieldsConnector.scala]: Good luck with metals formatting ;-)

File: project/metals.sbt (hunk 2 above)
Comment (AndySpaven) [reply project/metals.sbt]: Needs ignoring.

Comment (peteslater-ee) [reply project/metals.sbt]: Who me?

**Pull request merged**